automates/program_analysis/CAST2GrFN/visitors/cast_to_agraph_visitor.py

```diff
@@ -0,0 +1,449 @@
+        body = []
+        if(node.body != []):
+            body = self.visit_list(node.body)
+        node_uid = uuid.uuid4()
+        test_uid = uuid.uuid4()
+        body_uid = uuid.uuid4()
+
+        self.G.add_node(node_uid,label="Loop")
+        self.G.add_node(test_uid,label="Test")
+        self.G.add_node(body_uid,label="Body")
+
+        self.G.add_edge(node_uid,test_uid)
+        self.G.add_edge(test_uid,expr)
+        self.G.add_edge(node_uid,body_uid)
+        for n in body:
+            self.G.add_edge(body_uid,n)
+
+        return node_uid
+    
+    @visit.register
+    def _(self, node: ModelBreak):
+        """Visits a ModelBreak (break statment) node. 
+        The node's UID is returned"""
+        node_uid = uuid.uuid4()
+        self.G.add_node(node_uid, label="Break")
+        return node_uid
+
+    @visit.register
+    def _(self, node: ModelContinue):
+        """Visits a ModelContinue (continue statment) node. 
+        The node's UID is returned"""
+        node_uid = uuid.uuid4()
+        self.G.add_node(node_uid, label="Continue")
+        return node_uid
+
+    @visit.register
+    def _(self, node: ModelIf):
+        """Visits a ModelIf (If statement) node.
+        We visit the condition, and then the body and orelse 
+        attributes if we have any. They're all added to the Graph 
+        accordingly. The node's UID is returned."""
+        expr = self.visit(node.expr)
+        body = []
+        orelse = []
+        if(node.body != []):
+            body = self.visit_list(node.body) 
+        if(node.orelse != None):
+            orelse = self.visit_list(node.orelse)
+
+        node_uid = uuid.uuid4()
+        test_uid = uuid.uuid4()
+        self.G.add_node(node_uid,label="If")
+        self.G.add_node(test_uid,label="Test")
+        self.G.add_edge(node_uid,test_uid)
+        self.G.add_edge(test_uid,expr)
+        
+        body_uid = uuid.uuid4()
+        orelse_uid = uuid.uuid4()
+
+        self.G.add_node(body_uid,label="If Body")
+        self.G.add_node(orelse_uid,label="Else Body")
+
+        self.G.add_edge(node_uid,body_uid)
+        self.G.add_edge(node_uid,orelse_uid)
+
+        for n in body:
+            self.G.add_edge(body_uid,n)
+
+        for n in orelse:
+            self.G.add_edge(orelse_uid,n)
+
+        return node_uid
+
+    @visit.register
+    def _(self, node: ModelReturn):
+        """Visits a ModelReturn (return statment) node. 
+        We add the return value to this node with an edge.
+        The node's UID is returned"""
+        value = self.visit(node.value)
+        node_uid = uuid.uuid4()
+        self.G.add_node(node_uid, label="Return")
+        self.G.add_edge(node_uid, value)
+
+        return node_uid
+
+    @visit.register
+    def _(self, node: Module):
+        """Visits a Module node. This is the starting point for visiting
+        a CAST object. The return value isn't relevant here (I think)"""
+        return self.visit_list(node.body) 
```

Comment:
I think this PR might be missing your module/program level node additions. Just remember to add the module name onto the module node too!

Reply:
Piggy-backing off this ... @titomeister Can we include the name of the Module in the module node? Right now I believe the output has the module node named `Module: module`. Am I mistaken about that?

Reply:
You're right, it is missing. I must've not pushed it, but I wrote it in after our discussions. Sorry about that!
Paul: Right, the PDFs have names for both the Program and Module nodes. Right now they're both set to be the same name, the name of whatever is in the 'name' field of the Module node.